### PR TITLE
[BugFix] move SubfieldExprNoCopyRule from logical rule to physical rule

### PR DESCRIPTION
**Pull request by Youngwb**
Why I'm doing:
SubfieldExprNoCopyRule is in logical rewrite now, it can affect mv qury rewrite because of this rule will set copyflag for subfield, when mv plan and query plan have different pattern, the copyflag will be different
What I'm doing:
move SubfieldExprNoCopyRule from logical rule to physical rule
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/SubfieldExprNoCopyRule.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                rewriteProject(optExpression);
+            }
+            for (OptExpression input : optExpression.getInputs()) {
+                input.getOp().accept(this, input, context);
+            }
+            return optExpression;
+        }
+
+        public void rewriteProject(OptExpression optExpression) {
+            Projection projection = optExpression.getOp().getProjection();
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/SubfieldExprNoCopyRule.java:56]: commonSubProject？

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/SubfieldExprNoCopyRule.java:56]: Do not deal in this PR, only move origin logical

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/SubfieldExprNoCopyRule.java:58]: Done

**Review by Seaven — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by zombee0 — APPROVED** [review]

**Comment by Youngwb** [comment]
@Mergify backport branch-3.1-lake-optimize

**Comment by Youngwb** [comment]
@mergify backport branch-3.2

**Pull request merged**